options.go

```diff
@@ -0,0 +1,417 @@
+package httpsling
+
+import (
+	"encoding/base64"
+	"net/http"
+	"net/url"
+	"strings"
+	"unicode"
+
+	"github.com/ansel1/merry"
+	goquery "github.com/google/go-querystring/query"
+
+	"github.com/theopenlane/httpsling/httpclient"
+)
+
+// Option applies some setting to a Requester object
+type Option interface {
+	Apply(*Requester) error
+}
+
+// OptionFunc adapts a function to the Option interface
+type OptionFunc func(*Requester) error
+
+// Apply implements Option
+func (f OptionFunc) Apply(r *Requester) error {
+	return f(r)
+}
+
+// With clones the Requester, then applies the options to the clone
+func (r *Requester) With(opts ...Option) (*Requester, error) {
+	r2 := r.Clone()
+
+	err := r2.Apply(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return r2, nil
+}
+
+// MustWith clones the Requester, then applies the options to the clone
+func (r *Requester) MustWith(opts ...Option) *Requester {
+	if r2, err := r.With(opts...); err != nil {
+		panic(err)
+	} else {
+		return r2
+	}
+}
+
+// Apply applies the options to the receiver
+func (r *Requester) Apply(opts ...Option) error {
+	for _, o := range opts {
+		err := o.Apply(r)
+		if err != nil {
+			return merry.Prepend(err, "applying options")
+		}
+	}
+
+	return nil
+}
+
+// MustApply applies the options to the receiver
+func (r *Requester) MustApply(opts ...Option) {
+	if err := r.Apply(opts...); err != nil {
+		panic(err)
+	}
+}
+
+// Method sets the HTTP method
+func Method(m string, paths ...string) Option {
+	return OptionFunc(func(r *Requester) error {
+		r.Method = m
+		if len(paths) > 0 {
+			err := RelativeURL(paths...).Apply(r)
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
```

Comment:
can make this even simpler, e.g.
```suggestion
	        r.Method = m
		if len(paths) <= 0 {
			return nil
		}

		return RelativeURL(paths...).Apply(r)
```